flo_ai/models/flo_team.py

```diff
@@ -113,14 +119,42 @@ def build_team_supervisor_graph(self) -> FloTeam:
         # First add the nodes, which will do the work
         for flo_team_chain in flo_team_entry_chains:
             super_graph.add_node(flo_team_chain.name, self.get_last_message | flo_team_chain.chain | self.join_graph)
-        super_graph.add_node(self.flo_supervisor.name, self.flo_supervisor.executor)
+        super_graph.add_node(self.router.name, self.router.executor)
 
         for member in self.members:
-            super_graph.add_edge(member, self.flo_supervisor.name)
+            super_graph.add_edge(member, self.router.name)
 
-        super_graph.add_conditional_edges(self.flo_supervisor.name, self.router)
+        super_graph.add_conditional_edges(self.router.name, self.router_fn)
 
-        super_graph.set_entry_point(self.flo_supervisor.name)
+        super_graph.set_entry_point(self.router.name)
         super_graph = super_graph.compile()
         return FloTeam(self.name, super_graph)
+    
+    def build_linear_agents_graph(self) -> FloTeam:
+        flo_agent_nodes = [self.build_node(flo_agent) for flo_agent in self.flo_agents]
```

Comment:
We might have to see how we can apply the linear functional flo to teams, that is between teams. That is something we will have to implement here

Reply:
Making this change in a separate pull request